core/src/test/scala/integration/kafka/api/PlaintextConsumerSubscriptionTest.scala

```diff
@@ -244,6 +245,9 @@ class PlaintextConsumerSubscriptionTest extends AbstractConsumerTest {
     }, waitTimeMs = 5000, msg = "An InvalidTopicException should be thrown.")
 
     assertEquals(s"Invalid topics: [${invalidTopicName}]", exception.getMessage)
+    assertDoesNotThrow(new Executable {
```

Comment:
I like the idea of including the unsubscribe in the test, but with this we loose the coverage we initially had for subscribe(invalid) + close (if there's a regression and releaseAssignment stops ignoring the error we won't catch it). What about having 2 tests, mostly doing the same intially, but then one calls unsubscribe and asserts does not throw (like we have now), and the other one calls close explicitely and asserts the same, what do you think? (we could reuse most of the code for the 2 tests)

Reply:
Agree. Added test case `testSubscribeInvalidTopicCanClose` and `testSubscribeInvalidTopicCanUnsubscribe`. Thank you.